plugins/carla/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
+legacy_check()
```

Comment:
You don't need to use `legacy_check()` for new code - instead write the file for the new 3.0 CMake framework, but add checks for `OBS_CMAKE_VERSION` to guard old functionality (e.g. calling `setup_plugin_target` instead of `set_target_properties_obs`).

The `legacy_check` is reserved for existing code that hasn't been upgraded yet.

Reply:
is there an example I can follow? most plugins have this legacy step in there still.

Reply:
Alas there isn't because we haven't added a new plugin since - effectively you just guard the things that only work with the new system via `if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)` and do the legacy stuff in the `else` branch.

Reply:
handled in today's commits, a single file is in place now